core/vm/contracts.go

```diff
@@ -389,15 +389,24 @@ func (c *blake2F) RequiredGas(input []byte) uint64 {
 }
 
 const blake2FInputLength = 213
+const blake2FFinalBlockBytes = byte(1)
+const blake2FNonFinalBlockBytes = byte(0)
 
 var errBlake2FIncorrectInputLength = errors.New(
 	"input length for Blake2 F precompile should be exactly 213 bytes",
 )
 
+var errBlake2FIncorrectFinalBlockIndicator = errors.New(
+	"incorrect final block indicator flag",
+)
+
 func (c *blake2F) Run(input []byte) ([]byte, error) {
 	if len(input) != blake2FInputLength {
 		return nil, errBlake2FIncorrectInputLength
 	}
+	if input[212] != blake2FNonFinalBlockBytes && input[212] != blake2FFinalBlockBytes {
```

Comment:
Now you changed it to be super strict. Is that in line with the EIP? It wasn't when I checked yesterday, IIRC

Reply:
Yes, the EIP still says

> The boolean f parameter is considered as true if at least 1 bit out of the 8 bits is set.

Reply:
I am going to update EIP in an ~hour. I should probably push the EIP change first, sorry. Changed based on [this](https://github.com/ethereum/EIPs/issues/152#issuecomment-508494194) discussion.